visualization/app/codeCharta/core/tooltip/tooltipService.js

```diff
@@ -11,27 +10,61 @@ class TooltipService {
     /**
      * @constructor
      */
-    constructor(){
-
-        /**
-         * All available tooltips.
-         * @type {{Statements: Tooltip, RLOC: Tooltip, MCC: Tooltip}}
-         */
-        this.tooltips = {
-            Statements: new Tooltip("Statements", "Number of Statements"),
-            RLOC: new Tooltip("RLOC", "Real Lines of Code"),
-            MCC: new Tooltip("MCC", "MacCabe Complexity or cyclomatic complexity"),
-        };
+    constructor($rootScope, urlService){
+
+        this.tooltips = [];
+        var ctx = this;
+
+        urlService.getFileDataFromFile("tooltips.json").then(
+
+            //resolve
+            (data) => {
+                ctx.tooltips = data;
+                $rootScope.$broadcast("tooltips-changed", this.tooltips);
+            },
+            //reject
+            () => {
+                window.alert("error loading tooltips.json");
+            }
+
+        );
 
     }
 
     /**
      * returns the tooltip description related to the given key
      * @param {String} key
-     * @returns {string} description
+     * @returns {string} the description related to the given key
      */
     getTooltipTextByKey(key) {
-        return this.tooltips[key] ? this.tooltips[key].getTooltip() : "no description";
+
+        var patt = new RegExp(/_\S*_/);
```

Comment:
I have just added the regex explanation.
Const is a good idea in order to make code clearer, done.